lib/em-http/client.rb

```diff
@@ -182,7 +182,7 @@ def send_request(head, body)
       # Set the Content-Length if body is given,
       # or we're doing an empty post or put
       if body
-        head['content-length'] = body.bytesize
+        head['content-length'] = body.bytesize  if body.respond_to?(:bytesize)
```

Comment:
What is this guarding for?

Reply:
This is for passing non-String body values, e.g. `Pathname` or `IO`.  For `Pathname`, I should add setting content-length from Pathname#size.  For streaming `IO` objects, there's no way to know the size, and it's the responsibility of the caller to set content-length.

Reply:
Could you rebase this against https://github.com/igrigorik/em-http-request/pull/324?